### PR TITLE
[Feature][JsonSplit-api] modify API to Restful-01

### PR DESCRIPTION
**Pull request by marin-man**
#5498
This table records the modifications completed.

|name|API|UI||name|API|UI|
|:--:|:--:|:-:|:--:|:--:|:--:|:--:|
|ProcessDefinition| [1](https://github.com/apache/dolphinscheduler/pull/6016) | [1](https://github.com/apache/dolphinscheduler/pull/6016) ||AlertPluginInstance| [2](https://github.com/apache/dolphinscheduler/pull/6090) | [2](https://github.com/apache/dolphinscheduler/pull/6090) |
|ProcessInstances| [1](https://github.com/apache/dolphinscheduler/pull/6016) | [1](https://github.com/apache/dolphinscheduler/pull/6016) || DataSource|[2](https://github.com/apache/dolphinscheduler/pull/6090) | [2](https://github.com/apache/dolphinscheduler/pull/6090) |
|TaskDefinition| [1](https://github.com/apache/dolphinscheduler/pull/6016) | [1](https://github.com/apache/dolphinscheduler/pull/6016) ||Monitor| [3](https://github.com/apache/dolphinscheduler/pull/6100) |[3](https://github.com/apache/dolphinscheduler/pull/6100) |
|TaskInstances| [1](https://github.com/apache/dolphinscheduler/pull/6016) | [1](https://github.com/apache/dolphinscheduler/pull/6016) ||Queue| [3](https://github.com/apache/dolphinscheduler/pull/6100)  | [3](https://github.com/apache/dolphinscheduler/pull/6100) |
|Project| [1](https://github.com/apache/dolphinscheduler/pull/6016) | [1](https://github.com/apache/dolphinscheduler/pull/6016) ||Tenant| [3](https://github.com/apache/dolphinscheduler/pull/6100) | [3](https://github.com/apache/dolphinscheduler/pull/6100)  |
|AccessToken| [2](https://github.com/apache/dolphinscheduler/pull/6090) | [2](https://github.com/apache/dolphinscheduler/pull/6090) ||Scheduler| [3](https://github.com/apache/dolphinscheduler/pull/6100) | [3](https://github.com/apache/dolphinscheduler/pull/6100)  |
|AlertGroup| [2](https://github.com/apache/dolphinscheduler/pull/6090) | [2](https://github.com/apache/dolphinscheduler/pull/6090) ||Resource|4|4|
|WorkerGroup|4|4||UiPlugin|4|4|

[a document about modification](https://kdocs.cn/l/spOjvEzstRLk)

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java [hunk 1]
```diff
@@ -110,7 +102,7 @@
         @ApiImplicitParam(name = "locations", value = "PROCESS_DEFINITION_LOCATIONS", required = true, type = "String"),
         @ApiImplicitParam(name = "description", value = "PROCESS_DEFINITION_DESC", required = false, type = "String")
     })
-    @PostMapping(value = "/save")
```
Comment (brave-lee) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java:113]: Should this be kept ?

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java [hunk 2]
```diff
@@ -138,12 +130,12 @@ public Result createProcessDefinition(@ApiIgnore @RequestAttribute(value = Const
      * @param targetProjectCode target project code
      * @return copy result code
      */
-    @ApiOperation(value = "copy", notes = "COPY_PROCESS_DEFINITION_NOTES")
+    @ApiOperation(value = "batchCopyProcessDefinitions", notes = "COPY_PROCESS_DEFINITION_NOTES")
```
Comment (brave-lee) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java]: please check

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java (hunk 1 above)
Comment (marin-man) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java:113]: In the standard document, we use POST to represent create.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java (hunk 2 above)
Comment (marin-man) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java]: I have put it another way.
`@ApiOperation(value = "batchMoveByCodes", notes = "MOVE_PROCESS_DEFINITION_NOTES")`

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java [hunk 3]
```diff
@@ -58,15 +58,7 @@
...
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestAttribute;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.bind.annotation.ResponseBody;
-import org.springframework.web.bind.annotation.ResponseStatus;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java:61]: It is better not do this change, use import * is not a good practice.

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

**Review by marin-man — COMMENTED** [review]
These also need to import other packages, such as PutMapping and DeleteMapping.  I didn't change this part. The IDEA editor automatically changes to *.

**Comment by manmandm** [comment]
> These also need to import other packages, such as PutMapping and DeleteMapping. I didn't change this part. The IDEA editor automatically changes to *.

@ruanwenjun

**Comment by manmandm** [comment]
> These also need to import other packages, such as PutMapping and DeleteMapping. I didn't change this part. The IDEA editor automatically changes to *.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java [hunk 4]
```diff
@@ -81,7 +72,7 @@
  */
 @Api(tags = "PROCESS_DEFINITION_TAG")
 @RestController
-@RequestMapping("projects/{projectCode}/process")
+@RequestMapping("projects/{projectCode}/process-definitions")
```
Comment (wen-hemin) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java]: English singular is recommended.
project/{projectCode}/process-definition

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/TaskDefinitionController.java [hunk 5]
```diff
@@ -160,14 +153,14 @@ public Result queryTaskDefinitionVersions(@ApiIgnore @RequestAttribute(value = C
         @ApiImplicitParam(name = "code", value = "TASK_DEFINITION_CODE", required = true, dataType = "Long", example = "1"),
         @ApiImplicitParam(name = "version", value = "VERSION", required = true, dataType = "Int", example = "100")
     })
-    @GetMapping(value = "/version/switch")
+    @GetMapping(value = "/{code}/version/switch/{version}")
```
Comment (wen-hemin) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/TaskDefinitionController.java]: Get or Post ?

Get:
/{code}/version/{version}

Post:
/{code}/version/{version}/switch

**Review by wen-hemin — COMMENTED** [review]
Please check

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java (hunk 4 above)
Comment (marin-man) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java]: ok

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/TaskDefinitionController.java (hunk 5 above)
Comment (marin-man) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/TaskDefinitionController.java]: I think the first is good

**Review by gabry-lab — CHANGES_REQUESTED** [review]
please add the API Excel doc as an attachment

**Review by gabry-lab — APPROVED** [review]
+1

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/TaskDefinitionController.java (hunk 5 above)
Comment (gabry-lab) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/TaskDefinitionController.java]: /project/{code}/version/{version}/switch better

**Review by brave-lee — COMMENTED** [review]
+1

**Comment by brave-lee** [comment]
pls fix confilct first

**Review by lenboo — APPROVED** [review]
+1

**Pull request merged**